chia/wallet/wallet_state_manager.py

```diff
@@ -1478,6 +1478,14 @@ async def get_wallet_for_coin(self, coin_id: bytes32) -> Any:
         wallet = self.wallets[wallet_id]
         return wallet
 
+    async def clean_resync_tables(self) -> None:
+        async with self.db_wrapper.writer_maybe_transaction() as conn:
+            await conn.execute("DELETE FROM coin_record")
```

Comment:
Why not delete all tables? For example transaction records

Reply:
If we add new tables we will forget to delete them here in the future, guaranteed. I recommend using code that wipes everything all at once instead.

Reply:
That is the requirement, we want to save the offer and transaction records. Because we are doing this during the wallet running, drop table will cause "table not exist" errors.

Reply:
if we add a new table, that's also derived from the chain, we would need to delete it here too, as mariano points out. Can you think of a way to avoid it being forgotten? ideally some check that fails CI. Maybe an assert at the end of this function that only the tables we expect are left. That way the test for this function would fail if a new table is added.

Also, regarding `drop table` vs `delete from`. It really seems risky to do this while the wallet is running. Is there a good reason to not shut down the wallet node first, then drop the tables? There are so many scenarios and interleavings of logic that would have to be tested if we allow a running wallet to suddenly lose all entries in these tables.

Reply:
I tried drop table, it doesn't work while the connection is still open. The requirement is not shutdown the node. If we shutdown the wallet node, we need a daemon or some external thing to recreate a node.

Reply:
when you say "connection", do you mean the database connection?
How did it fail? 
It seem like that failure could be an indication of some additional synchronization that needs to happen, and your solution of just removing all the rows is just masking an underlying problem.

Reply:
Here are the reasons I don't drop tables:
1. It's hard to reuse the existing code to recreate tables. Fully restarting the wallet can solve the problem but it is against the requirement.
2. It will throw exceptions while the wallet is still running. The insert will fail if it happens after we delete the table and before we create the new table.